snakemake/Snakefile

```diff
@@ -116,21 +117,31 @@ rule guidescan_index:
 rule generate_kmers:
     input: rules.download_sequence.output[1],
             rules.install_guidescanpy.output
-    output: f"{OUTPUT_DIR}/kmers/{{organism}}.csv"
+    output: f"{OUTPUT_DIR}/kmers/{{enzyme}}/{{organism}}.csv"
+    params: pam = lambda wildcards: config['pams'][wildcards.enzyme]
     shell:
         """
-        guidescanpy generate-kmers {input[0]} --max-kmers 100 > {output}
+        guidescanpy generate-kmers --max-kmers 10000 {input[0]} --pam {params.pam} > {output}
     """
 
 
 rule guidescan_enumerate:
     input: rules.guidescan_index.output,
             rules.generate_kmers.output
-    output: f"{OUTPUT_DIR}/bam_files/{{organism}}.bam"
-    params: index_file = f"{OUTPUT_DIR}/index/{{organism}}.fna.index"
+    output: f"{OUTPUT_DIR}/databases/{{enzyme}}/{{organism}}.sam"
+    params: index_file = f"{OUTPUT_DIR}/indices/{{organism}}/{{organism}}.fna.index",
     shell:
         """
+        guidescan enumerate {params.index_file} --kmers-file {input[1]} --output {output} --format sam
+        """
 
-        guidescan enumerate {params.index_file} --kmers-file {input[1]} --output {output}
-    """
+
+rule add_sam_ce:
+    input: rules.guidescan_enumerate.output
+    output: f"{OUTPUT_DIR}/databases/{{enzyme}}/{{organism}}.bam.sorted"
+    shell:
+        """
+        guidescanpy add-sam-tag {input} --ce -o {output}
+        python -c "import pysam; pysam.sort('-o', '{output}', '{output}'); pysam.index('{output}')"
```

Comment:
Let's create an `{{organism}}.bam` first, and then sort it and save as `{{organism}}.bam.sorted`